### PR TITLE
feat: validate new project name (#237)

### LINKED ISSUE #237: Add name validator for New project dialog
[kind/Enhancement]

## Which functionality do you think we should update/improve?
Add validator for New Project dialog:
image
VSCode:
image

## Why is this needed? 
This is implemented in VSCode and JBT.

### PR DESCRIPTION
**Pull request by adietish**
fixes #237, #674, #701
depends on #661

### CODE REVIEW

**Comment by sbouchet** [comment]
Same remark as #689 . is it possible to change labels if the cluster is openshift(project) or kubernetes(namespace) ?

**Comment by sbouchet** [comment]
> Same remark as #689 . is it possible to change labels if the cluster is openshift(project) or kubernetes(namespace) ?

created #701 for fixing actions labels only.

**Comment by adietish** [comment]
I'll fix #701 in this PR as providing different labels for kubernetes/openshift is closely related to the actions to launch it.

I investigated a unified approach where where `TreeAction#update` would set the Action label in the ctx menu. It doesn't work. Weird enough, `update` is called before the context menu is to be shown for the first time, but it has no effect. It takes only effect on the 2nd, 3rd, 4th, etc. call. 
What would have worked was to register an alternative text via `AnAction#addTextOverride` in the constructor. Unfortunately I need an `AnActionEvent` to get the root node with the odo that would tell me if we have an openshift or a kubernetes cluster (`Odo.isOpenShift()`).
This left me with the only approach where there are 2 distinct actions for openshift and kubernetes cluster that would be visible for the specified cluster type.

**Comment by adietish** [comment]
@sbouchet: please re-review.

**Comment by adietish** [comment]
/override ci/prow/e2e-openshift  tide "Aggregated Test Report"

File: src/main/java/org/jboss/tools/intellij/openshift/actions/project/CreateProjectAction.java [hunk 1]
```diff
@@ -19,18 +19,43 @@
...
+
+  @Override
+  public void update(AnActionEvent e) {
+    super.update(e);
+    if (e.getPresentation().isVisible()) {
+      Odo odo = getOdo(e);
+      if (odo == null) {
+        return;
+      }
+      if (!odo.isOpenShift()) {
```
Comment (sbouchet) [inline src/main/java/org/jboss/tools/intellij/openshift/actions/project/CreateProjectAction.java:53]: this is redundant with getKind() that does the same test.

**Comment by adietish** [comment]
> there is no update on ChangeActiveProjectAction, thus the label is always 'project'.

I cannot replicate this. Things work fine for me when switching from kind to crc.

https://github.com/redhat-developer/intellij-openshift-connector/assets/25126/63958103-7872-4d92-ab37-40560b999e58


What am I missing?

**Comment by sbouchet** [comment]
> @sbouchet:
> 
> > there is no update on ChangeActiveProjectAction, thus the label is always 'project'.
> 
> I cannot replicate your issue. Things are working fine for me when switching from kind to crc.
> 
>  Screen.Recording.2024-02-08.at.20.17.39.mov 
> What am I missing?

look at 0:13. this is not the "create" label, but the "Change " one that is always displaying "Project". there is a missing update() method in the ChangeAction.

**Comment by adietish** [comment]
> > @sbouchet:
> > > there is no update on ChangeActiveProjectAction, thus the label is always 'project'.
> > 
> > 
> > I cannot replicate your issue. Things are working fine for me when switching from kind to crc.
> > Screen.Recording.2024-02-08.at.20.17.39.mov
> > What am I missing?
> 
> look at 0:13. this is not the "create" label, but the "Change " one that is always displaying "Project". there is a missing update() method in the ChangeAction.

This PR is about the "New Project" wizard, not the "Change Project" wizard :)
So this is a different issue, I'll file and fix it.

**Comment by adietish** [comment]
@sbouchet 👍 gotcha, this was included in your PR for #701. Adding it here, too.

**Comment by adietish** [comment]
@sbouchet please re-review, thanks!

**Comment by sbouchet** [comment]
Capture d’écran du 2024-02-09 12-12-47
left two places with "project" when using kubernetes cluster. 
- one on validation message on dialog 
- one on missing project/namespace message on tree

**Comment by adietish** [comment]
@sbouchet fixes those 2, please re-review.

**Comment by adietish** [comment]
/override ci/prow/e2e-openshift "Aggregated Test Report" tide

**Pull request merged**